src/engraving/api/v1/score.cpp

```diff
@@ -274,33 +262,14 @@ QQmlListProperty<System> Score::systems()
     return wrapContainerProperty<System>(this, score()->systems());
 }
 
-/** APIDOC @property {boolean} - has lyrics */
-bool Score::hasLyrics() const
-{
-    return score()->hasLyrics();
-}
-
-/** APIDOC @property {number} - count of lyrics */
-int Score::lyricCount() const
-{
-    return score()->lyricCount();
-}
-
-/** APIDOC @property {Lyric[]} - list of lyrics */
-QQmlListProperty<Lyric> Score::lyrics() const
-{
-    static std::vector<Lyrics*> list = score()->lyrics();
-    return wrapContainerProperty<Lyric>(this, list);
-}
+//---------------------------------------------------------
+//   Score::spanners
+//---------------------------------------------------------
 
-/** APIDOC
-* Extracts all lyrics in the score and returns them in a single string.
-* @method
-* @return {string} - lyrics string
-*/
-QString Score::extractLyrics() const
+QQmlListProperty<EngravingItem> Score::spanners()
 {
-    return score()->extractLyrics();
+    m_cachedSpannerList = score()->spannerList();
```

Comment:
I think it's easier with a static variable, like here 
https://github.com/musescore/MuseScore/blob/master/src/engraving/api/v1/score.cpp#L292